File: Backend/src/main/java/unical/demacs/rdm/persistence/dto/JsonDTO.java
package unical.demacs.rdm.persistence.dto;

import lombok.Data;
import java.util.List;

@Data
public class JsonDTO {
    private List<JobDTO> jobs;
    private List<MachineDTO> machines;
    private List<MachineTypeDTO> machineTypes;
    private List<ScheduleDTO> schedules;
}

File: Frontend/src/app/generated-api/model/jsonDTO.ts

import { JobDTO } from './jobDTO';
import { MachineDTO } from './machineDTO';
import { MachineTypeDTO } from './machineTypeDTO';
import { ScheduleDTO } from './scheduleDTO';

export interface JsonDTO {
    jobs?: Array<JobDTO>;
    machines?: Array<MachineDTO>;
    machineTypes?: Array<MachineTypeDTO>;
    schedules?: Array<ScheduleDTO>;
}
